CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+* release/v1.x, (for example [release/v1.5](https://github.com/rancher/rke/tree/release/v1.5)
+* release/v1.(x-1) (for example [release/v1.4](https://github.com/rancher/rke/tree/release/v1.4)
+
+Each track releases at the same time an RKE binary. Each release of RKE officially supports a few Kubernetes versions. Some versions are shared by both tracks but normally the track with the higher minor version includes at least one newer Kubernetes version. For example, as it can be observed in [release v1.4.11](https://github.com/rancher/rke/releases/tag/v1.4.11), the following Kubernetes versions are supported:
+* v1.26.9-rancher1-1
+* v1.25.14-rancher1-1
+* v1.24.17-rancher1-1
+* v1.23.16-rancher2-3
+
+whereas [release v1.5.0](https://github.com/rancher/rke/releases/tag/v1.5.0) includes:
+* v1.25.14-rancher1-1
+* v1.26.9-rancher1-1
+* v1.27.6-rancher1-1
+
+By support, it means we can deploy a cluster with that Kubernetes version using the binray included in that RKE release.
+
+When doing a commit in RKE code, we must carefully think if we want that commit in both tracks and hence backported. Sometimes, there is a reason to no backport it, for example a flag, that we would like to limit to only the newer track. However, that is exceptional and normally commits end up in both branches as a way to reduce the differences between those and simplify RKE maintainability. Therefore, usually we backport a commit between both RKE tracks. In that case, the following steps should be applied:
+
+* Create an Issue in rancher/rancher github project for Project Managers and QA to track the commit. Set the decided milestone for release/v1.x in the issue
+* Create another Issue in the same github project and in the description, specify it is a backport of the previous commit. Set the decided milestone for release/v1.(x-1) in the issue
+* Create the PR with the commit in branch release/v1.x and in the description, link the newly created issue
+* Cherry-pick the commit in branch release/v1.(x-1) and in the description, link the created issue for that RKE track.
+
+A commit is normally available in all supported Kubernetes versions of the release unless scoped in the code to a specific version. Here are two examples for this:
+* Available in all supported Kubernetes versions: https://github.com/rancher/rke/commit/1dad2582e1efe5564b6d1a7385afad70d5dbd1f7
+* Scoped to a specific version: https://github.com/rancher/rke/commit/feca6a3048c1f8c9031ab2849a46c46a7ea1d673
+
+Note that in the scoped case, there is a check on the version, which is normally fetch by executing the line `util.StrToSemVer(k8sVersion)`. In this case, as the same code is executed for all Kubernetes versions, unless scoped, there is no need to create extra PRs to effectively backport the code to all Kubernetes versions.
```

Comment:
I'd add a link to the code that checks the version

Reply:
You mean github.com/coreos/go-semver/semver? We are using that type and call functions such as https://github.com/rancher/rke/blob/release/v1.5/cluster/validation.go#L752

Reply:
Or this https://github.com/rancher/rke/blob/c56fb8c42a079d569dd402c1243a5ae16b4004c9/cluster/validation.go#L748 but yes